src/test/ui/pattern/issue-88074-pat-range-type-inference-err.rs

```diff
@@ -0,0 +1,15 @@
+trait Zero {
+    const ZERO: Self;
+}
+
+impl Zero for String {
+    const ZERO: Self = String::new();
```

Comment:
When associated consts are involved in this error, we should likely point at them.